src/performRequest.js

```diff
@@ -0,0 +1,161 @@
+const defaultRequest = require('request');
+const caseless = require('caseless');
+
+const defaultLogger = require('./logger');
+
+
+/**
+ * Performs the HTTP request as described in the 'transaction.request' object.
+ *
+ * In future we should introduce a 'real' request object as well so user has
+ * access to the modifications made on the way.
+ *
+ * @param {string} uri
+ * @param {Object} transactionReq
+ * @param {Object} [options]
+ * @param {Object} [options.logger] Custom logger
+ * @param {Object} [options.request] Custom 'request' library implementation
+ * @param {Object} [options.http] Custom default 'request' library options
+ * @param {Function} callback
+ */
+function performRequest(uri, transactionReq, options, callback) {
+  if (typeof options === 'function') {
+    [options, callback] = [{}, options];
+  }
+  const logger = options.logger || defaultLogger;
+  const request = options.request || defaultRequest;
+
+  const httpOptions = Object.assign({}, options.http || {});
+  httpOptions.proxy = false;
+  httpOptions.followRedirect = false;
+  httpOptions.encoding = null;
+  httpOptions.method = transactionReq.method;
+  httpOptions.uri = uri;
+
+  try {
+    httpOptions.body = getBodyAsBuffer(transactionReq.body, transactionReq.bodyEncoding);
+    httpOptions.headers = setContentLength(transactionReq.headers, httpOptions.body);
```

Comment:
I think that `setContentLength` is a little bit misleading here, `normalizeHeaders` or something like that would be more fitting in this case.

Reply:
It doesn't interfere with any other headers. It detects the name of the content-length header if it exists, and sets it to the correct value. In case it doesn't exists, it adds the header. IMHO the name is right.

Reply:
🤔

Reply:
1f6c251